### ISSUE TITLE
[Feature] API Tokens

### ISSUE THREAD

**Issue by MustafaAmer-1** [mikesprague/clean-start#29]
first of all thanks for this awesome work! 
I am really like this extension, I have started using it about year ago and it's really helpful.
I suggest if you can make a something like build server that other developers can use to build and test the extension on it without having to get the API token keys required from font awesome token, Dark Sky, Unsplash, and Google Maps.
I think this will also help people make their own customized version of the extension, I have many edits I need to do to the extension that will make my life easier.
Or if there is away to get these tokens for free (especially font awesome token) please let me know.
Thanks!

**Comment by mikesprague** [comment]
@MustafaAmer-1 

Thanks for the suggestions.

It has always bothered me that the "free" font-awesome still needs an API key to use their packages. I think I can replace those with a different icon solution and will try to make some time to do that soon. This would remove the need for a font-awesome API key.

Regarding the others.... 

Dark Sky will be shutting down at the end of this year and that will service need to be replaced. Not sure if I will replace it or just remove the weather as there are very few solutions that offer the same amount of data and such a generous free tier. Feel free to offer any suggestions but if I do use another weather service, I would anticipate still needing API keys for myself and any other devs else using it. Again, open to suggestions and/or pull requests using other weather services(s).

Unsplash and Google Maps could each ban me if I proxied their APIs to others (even just for dev purposes). However, they each offer API keys for free so the only real headache there is the time to sign up for and create your own.  

I'll leave this open until I remove font-awesome.

**Comment by mikesprague** [comment]
On the off chance this is still of interest to you...

I made some significant changes (that break the currently published extension until Google approves my updated submission) but that mostly address the previous issues:

- there is no longer a need for a FontAwesome API key
- Dark Sky has been converted to OpenWeatherMap (which have free API keys that support 1 million requests per day, links to sign up in README)
- there is still also a need for Unsplash API key and secret but those also remain free (and now have sign up links in README)

Unrelated to your asks about API keys, the build system has been refactored from Webpack to Vite and the hosting has been moved from Vercel to CloudFlare Pages (using Wrangler CLI along with Vite for local dev).

As previously mentioned, there's no viable way for me to provide any proxied API keys without putting my accounts at risk of breaking Terms of Services and losing access.

I think streamlining things to only need the above keys, and using providers that let you sign up for free (and without asking for any payment info), should eliminate most barriers to development.
